org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl

```diff
@@ -69,14 +71,23 @@ import [javaClassFullNameForAdaptorManager(anAdaptorInterface) /];
 // [protected ('imports')]
 // [/protected]
 
+/**
+ * During the initialization of this ServletListener, the base URI for the OSLC resources produced by this server is configured through the OSLC4J method setPublicURI().
+ * A number of approaches can be used to set this base URI:
+ * 1. Set a system environment property of name "LYO_BASE".
+ * 2. In the web.xml file, set a "context-param" property, with name "[javaClassPackageNameForAdaptorServletListener(anAdaptorInterface) /].baseurl"
+ * 3. Override the value of the DEFAULT_BASE constant.
+ */
 public class [javaClassNameForAdaptorServletListener(anAdaptorInterface) /] implements ServletContextListener  {
-    private static final String DEFAULT_BASE = "http://localhost:8080";
-    private static final String PROPERTY_BASE = servletContextParameterName("baseurl");
-    private static final Logger logger = Logger.getLogger([javaClassNameForAdaptorServletListener(anAdaptorInterface) /].class.getName());
 
-    //If you are using another servletName in your web.xml configuration file, modify this variable early in the method contextInitialized below
+    //These are default names. You can modify any of these "constants" early in the contextInitialized() method.
+    private static String BASE_PATH_ENV_KEY = "LYO_BASE";
```

Comment:
As these are no longer constant, I think they can be inlined in a method as local variables.

Reply:
I thought it better to collect all variables at the start, so that they are (1) visible to programmer (2) They can be overriden at the start of the init method.

Reply:
Yes, but at the start of the method, not the class.

Reply:
Fair enough. Fixed.